extensions/ql-vscode/src/extension.ts

```diff
@@ -493,6 +494,29 @@ async function activateWithInstalledDistribution(
     }
   }
 
+  async function previewQueryHelp(
+    selectedQuery: Uri
+  ): Promise<void> {
+    const path = selectedQuery ? selectedQuery.path : window.activeTextEditor!.document.uri.fsPath;
+    if(path) {
+      // Create temporary directory
+      const tmpDir = tmp.dirSync();
```

Comment:
Using a temporary directory is sensible. However we must be extra careful to clean it up properly, especially on Windows.

Have you seen the other uses of temporary directories in `run-queries.ts`? I suggest you follow that pattern as closely as possible. In particular, I recommend passing similar options to `dirSync`, and creating a dispose callback that can be registered with the extension via `ctx.subscriptions.push(callback)`, so that the extension cleans up the temporary directories when it shuts down.

---
More detailed discussion: We could achieve more elegant code by adding an extra field to the `QueryInfo` class in `run-queries.ts`, which currently tracks a `dilPath` and `resultsPath` for each completed query. What do others think?
- Advantages: It handles all the temporary directories already, associates each query with a number of temporary file paths, and doesn't repeat work if the file has already been generated.
- Disadvantages: It only works for queries that have been run, which seems unreasonable for previewing query help. We would also have to track whether the `.qhelp` file had been modified.